BUILD.bazel

```diff
@@ -363,6 +363,7 @@ cc_library(
         "@com_github_grpc_grpc//:grpc++",
         "@com_google_absl//absl/container:flat_hash_map",
         "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_absl//absl/synchronization",
```

Comment:
I guess we can remove this now? (if this was added for the Reader/Writer lock)

Reply:
Still needed for the `absl::Mutex` and the `absl::MutexLock`, I believe.

Reply:
Ah, I see. I thought it was already added somewhere since we used locks everywhere, but it looks like it was included in `ray_util`

Reply:
We're often relying on implicit linking via transitive dependencies for these kind of libraries, which makes a lot of these Bazel targets brittle when it comes to refactoring (e.g., if the `ray_util` dependency was removed, the `ray_common` library would break since it was implicitly relying on libraries included by `ray_util`). IMO, if a library (Bazel target) has a dependency like `absl/synchronization`, it should explicitly have it as a Bazel dependency.